frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -298,6 +293,34 @@ dbg:guard(FileManagerMenu, 'setUpdateItemTable',
         end
     end)
 
+function FileManagerMenu:exitOrRestart(callback)
+    if SetDefaults.settings_changed then
+        UIManager:show(ConfirmBox:new{
+            text = _("You have unsaved default settings. Save them now?\nTap \"Cancel\" to return to KOReader."),
+            ok_text = _("Yes"),
```

Comment:
Btw, https://github.com/koreader/koreader/commit/1c25d0fd1b4d19d8da7f5144f2ebf75ac4888cb7

So the dialog should look like this:

<kbd>Cancel</kbd> <kbd>Don't save</kbd> <kbd>Save</kbd>